common/spec/dependabot/config/ignore_condition_spec.rb

```diff
@@ -121,6 +124,53 @@ def expect_ignored(versions)
         it "ignores expected versions" do
           expect_allowed(minor_upgrades)
           expect_ignored(patch_upgrades + major_upgrades)
+          expect_allowed([dependency_version])
+        end
+      end
+
+      context "with a `major.minor.patch.patch` semver version" do
+        let(:dependency_version) { "1.2.3.1" }
+
+        context "with ignore_major_versions" do
+          let(:update_types) { ["version-update:semver-major"] }
+
+          it "ignores expected versions" do
+            expect_allowed(patch_upgrades + minor_upgrades)
+            expect_ignored(major_upgrades)
+            expect_allowed([dependency_version])
+          end
+
+          it "returns the expected range" do
+            expect(ignored_versions).to eq([">= 2.a, < 3"])
+          end
+        end
+
+        context "with ignore_minor_versions" do
+          let(:update_types) { ["version-update:semver-minor"] }
+
+          it "ignores expected versions" do
+            expect_allowed(patch_upgrades + major_upgrades)
+            expect_ignored(minor_upgrades)
+            expect_allowed([dependency_version])
+          end
+
+          it "returns the expected range" do
+            expect(ignored_versions).to eq([">= 1.3.a, < 2"])
+          end
+        end
+
+        context "with ignore_patch_versions" do
+          let(:update_types) { ["version-update:semver-patch"] }
+
+          it "ignores expected updates" do
+            expect_ignored(["1.2.3.2", "1.2.4.0"])
```

Comment:
```suggestion
            expect_ignored(patch_upgrades + ["1.2.3.2", "1.2.4.0"])
```

Idea: include the base cases.